spec/index.bs

```diff
@@ -3548,6 +3529,74 @@ integration such as source-language debugging.
 
 ### Shader Module Compilation Information ### {#shader-module-compilation-information}
 
+<script type=idl>
+enum GPUCompilationMessageType {
+    "error",
+    "warning",
+    "info"
+};
+
+[Exposed=Window, Serializable]
+interface GPUCompilationMessage {
+    readonly attribute DOMString message;
+    readonly attribute GPUCompilationMessageType type;
+    readonly attribute unsigned long long lineNum;
+    readonly attribute unsigned long long linePos;
+    readonly attribute unsigned long long spanOffset;
+    readonly attribute unsigned long long spanLength;
```

Comment:
I'd like to have naming that doesn't associate the "length" with either mode in particular. Spitballing:`startLineNum`, `startLinePos`, `startOffset`, `length`?

Reply:
I have a minor preference for not changing the existing names if possible, and I don't know that specifying `start_` helps them much. I did rename the `spanOffset` and `spanLength` to simply `offset` and `length`, though.

Reply:
These names lgtm.